### PR TITLE
Ready to review: Implement TAK Quickstart Guide & TAK Usage Guide

### PR DESCRIPTION
**Pull request by benjamsf**
### Ready to review: 17 Oct
# Why
As per 1.0 scoping (task: https://github.com/pvarki/rasenmaeher-ui/issues/1), we will be implementing TAK related service material directly to Rasenmaeher UI. Original html fragment design to be implemented1.1. -> later. 

This PR:
- does the visual components
- implements: TAK quickstart guide & usage guide for Android/iOS/Win & integrates them to routes
- alters: adminview and soldierview to hold TAK as a foldable card (IDEA: Later on, other services served by certain RASENMAEHER deployment would appear as additional cards there. Typical, 1-3 cards).
- aims to make Rasenmaeher UI look nice with mobile & desktop and be the 1.0 visual implementation
After this, we need enrollment logic implemented & add visuals to those views. 

# What
## General visual things
Add visual components to ./components and alter core components:
- Alter Layout. It is now called at each view, has props for whether to show Navbar or Header, navbartitle, heroimage, backUrl
- Add Footer that is propped true/false in Layout, default True
- Add in components: Card, CardContainer, ServiceInfo and ServiceUsage cards, foldable cards (Unfoldable, foldable, button foldable), statusbar, Text, textcard, GuideSection

## Homeviews
- Alter homepage:
- - despite Figma design it is actually rather tedious to navigate through multiple pages to get your stuff, so now services would list as unfoldable cards to Homeview
- - Remove ServicesView
- - Add TakService component that is a card that is called to Homeview now (TO DO: html fragment design instead of integrated to rasse)
- - In the html fragment future, could they be loaded as text+images under these components? So when a new product integration is to be done, probably only minimal new styling to RM UI and then the content provided by RM-newproduct-AP
- Add respective view components, nested ./components/views/servicetak/ - usage and quickstart

## Usage & Quickstart views
- Add components specific to this usage case we ship at 1.0 - they are at ./components/views/servicetak/usage/
- Implement TAK quickstart guide for Android, iOS, Windows
- - has a quickstartview that asks user which device type
- - and respective vier for each
- Implement TAK usage guide for Android, iOS, Windows
- - Logic: The contents could change, the format should stay
- - In the UsageView there is top-level information on how-to
- - From UsageView, how-to do that with each device **(iTAK missing yet)**

## Other
- Add necessary images
- Add routing

# Images of changes
Näyttökuva 2023-10-17 kello 16 30 36
Näyttökuva 2023-10-17 kello 16 30 43
Näyttökuva 2023-10-17 kello 16 30 52
Näyttökuva 2023-10-17 kello 16 31 02
Näyttökuva 2023-10-17 kello 16 31 18
Näyttökuva 2023-10-17 kello 16 31 26
Näyttökuva 2023-10-17 kello 16 37 03
Näyttökuva 2023-10-17 kello 16 37 59

# What next
- How to manage user groups (admin, fighter) are not? implemented yet, need to implement handling that logic
- Need to implement how TAK client zips gets to be available from ServiceTakView button "Hae viestiperusteesi"
- Fix ESLint errors elsewhere
- Consider html fragment design for anything related to some Service, by 1.1 or later
- **Add usageguide/iTAK**, waiting for MikkoH
- Add these visuals to any enrolling/user management related screens, I can do that while eskimo does the realwork

### CODE REVIEW

File: src/Router.tsx
Comment (3zk1m0) [inline src/Router.tsx:1]: TAK routes should be defined in own file. 
Other parts of the app could be grouped later.

File: src/components/Card.tsx [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
   children?: ReactNode; 
 }
 
-export function Card({ title, image, url }: { title: string; image?: string, url?:string }) {
+export function Card({ title, details, image, url }: CardProps) {
     return (
       <a
         href={url}
-        className="relative bg-backgroundLight rounded-lg w-full h-36 md:h-52 lg:h-72 mb-5 overflow-hidden"
+        className="relative bg-backgroundLight rounded-lg w-full w-[380px] max-w-full h-17 md:h-17 lg:h-36 m-2 overflow-hidden flex items-center justify-center"
```
Comment (3zk1m0) [inline src/components/Card.tsx:16]: Constant width (w-[380]) could cause problems with some devices.

Comment (benjamsf) [reply src/components/Card.tsx:16]: Got some of these there, will do them alternatively

File: src/components/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      </div>
+
+      <hr className="mx-auto w-56" />
+
+      <div className="py-5 text-xs">
+        {parser("Proudly served by PVATK")}
+        <br />
+        {parser("&copy; Puolustusvoimat")}
+        <br />
+        {parser("Yhteys: <p class='underline'>helpdesk@pvarki.fi</p>")}
```
Comment (3zk1m0) [inline src/components/Footer.tsx:20]: What parser provides here that requires it?

File: src/components/NavigateButtons.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    navigate(forwardUrl);
+    setContinueClicked(false);
+  };
+
+  return (
+    <div className="fixed bottom-2 left-0 right-0 flex justify-center">
+      <section className="items-start rounded bg-zinc-900 flex w-[420px] min-h-[75px] max-w-full gap-2.5 mt-24 p-2.5 max-md:justify-center">
+        <Button
+          styling={`transform transition-transform duration-200 justify-center items-center rounded ${BackClicked ? "scale-95" : "scale-100"} self-stretch flex min-w-[150px] flex-col px-6 py-3 ${BackClicked ? "bg-blue-900" : "bg-zinc-800"}`}
+          onMouseDown={() => setBackClicked(true)}
```
Comment (3zk1m0) [inline src/components/NavigateButtons.tsx:34]: onPointerDown would also allow touch events

File: src/components/ServiceUsageCard.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  children?: React.ReactNode;
+}
+
+export function ServiceUsageCard({ children }: ServiceUsageCardProps) {
+  return (
+    <div>
+      <CardsContainer>
+        <ButtonFoldableCard title="Käyttömalli">
+          <div className="flex flex-col items-center justify-center p-5">
+
```
Comment (3zk1m0) [inline src/components/ServiceUsageCard.tsx:14]: Empty lines?

File: src/components/StatusBar.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    progressNow: number;
+}
+
+export function StatusBar(props: StatusBarProps) {
+    return (
+        <div className="flex flex-col items-center space-y-4">
+            <h1 className="text-xl font-bold text-white">{props.title}</h1>
+            <div className="flex items-center">
+                {Array.from({ length: props.progressMax }).map((_, index) => (
+                    <React.Fragment key={index}>
```
Comment (3zk1m0) [inline src/components/StatusBar.tsx:15]: <></> short hand for fragments, 
But doesn't this work without out it, just one div?

File: src/components/Card.tsx (hunk 1 above)
Comment (benjamsf) [reply src/components/Card.tsx]: true, fix'd

File: src/components/TextCard.tsx [hunk 6]
```diff
@@ -11,12 +11,12 @@ export function TextCard({
   details?: string; 
 }) {
   return (
-    <div className="flex flex-col items-center justify-center bg-backgroundLight rounded-lg w-full h-36 md:h-52 lg:h-72 mb-5 overflow-hidden">
+    <div className="flex flex-col items-center justify-center bg-backgroundLight rounded-lg w-full w-[380px] max-w-full h-36 md:h-36 lg:h-36 mb-5 overflow-hidden">
```
[the remark this answers is not in the document]
Comment (benjamsf) [reply src/components/TextCard.tsx]: fixed

File: src/components/ServiceUsageCard.tsx (hunk 4 above)
Comment (benjamsf) [reply src/components/ServiceUsageCard.tsx:14]: Forgot, had something else there at one point -done!

File: src/components/StatusBar.tsx (hunk 5 above)
Comment (benjamsf) [reply src/components/StatusBar.tsx:15]: yeah, actually does, overcomplicated & hence simplified

File: src/components/Footer.tsx (hunk 2 above)
Comment (benjamsf) [reply src/components/Footer.tsx:20]: Well actually, nothing. But guess who used parser in the GuideSection component, and then wrote 2 x 3 guides and supplied all formatting to those written things with html tags?

Comment (benjamsf) [reply src/components/Footer.tsx:20]: removed this one here though

**Comment by benjamsf** [comment]
All things fixed, except potentially using html-react-parser in supplied strings to the component GuideSection, at QuickstartGuide and UsageGuide

**Comment by 3zk1m0** [comment]
It seems like Prettier hasn't been executed, or our configuration might be too permissive.

**Comment by benjamsf** [comment]
I'll fix them, anything else in general that you notice? Good practice for me

**Pull request closed without merging**